cros_gralloc/gralloc4/CrosGralloc4Mapper.cc

```diff
@@ -149,16 +150,9 @@ Return<Error> CrosGralloc4Mapper::validateBufferSize(void* rawHandle,
     }
 
     PixelFormat crosHandleFormat = static_cast<PixelFormat>(crosHandle->droid_format);
-#ifdef USE_GRALLOC1
-    int32_t yuvFormat = static_cast<int32_t>(descriptor.format);
-    if (descriptor.format != crosHandleFormat && yuvFormat != crosHandle->droid_format &&
-        !(descriptor.format == PixelFormat::YCBCR_420_888 &&
-            crosHandle->droid_format == HAL_PIXEL_FORMAT_NV12)) {
-        drv_log("Failed to validateBufferSize. Format mismatch.\n");
-#else
-    if (descriptor.format != crosHandleFormat) {
+
+    if (descriptor.format != crosHandleFormat && !is_flex_format(cros_gralloc_convert_format((int32_t)descriptor.format))) {
```

Comment:
When allocate HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED, we treat it as DRM_FORMAT_XBGR8888. Then when we validate, can we also check HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED and DRM_FORMAT_XBGR8888 are matched.

Reply:
yes, true, this will be more rigorous.